spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2521,9 +2524,12 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
             new SparkToParquetSchemaConverter(conf).convert(scan.requiredSchema)
           val dataSchemaParquet =
             new SparkToParquetSchemaConverter(conf).convert(scan.relation.dataSchema)
+          val partitionSchemaParquet =
+            new SparkToParquetSchemaConverter(conf).convert(scan.relation.partitionSchema)
```

Comment:
#1103 discusses how the schemas have already lost necessary information at this point. Should we construct a new partition schema from the true Parquet schema rather than the partitionSchema that may have lost/converted type information already?

Reply:
This copies from existing code.

Actually, I can just convert the Spark schema to Arrow types in JVM and serialize it to native side. I did similar thing in shuffle writer. Then we won't lose any information.